frontend/src/lib/components/BranchButton.svelte

```diff
@@ -75,6 +76,25 @@
 		);
 	}
 
+	// Fetches the default branch name from the backend and sets it to baseBranch store
+	async function fetchDefaultBranch() {
+		try {
+			const response = await fetch(`${apiAddress}/api/repos/default-branch`);
+
+			if (response.ok) {
+				const data = await response.json();
+				const defaultBranch = data.data;
+
+				// Set the default branch to the baseBranch store
+				baseBranch.set(defaultBranch);
+			} else {
+				console.error('Failed to fetch default branch:', response.statusText);
+			}
+		} catch (error) {
+			console.error('Error fetching default branch:', error);
+		}
```

Comment:
I would prefer if try/catch wasn't used here, it would only really apply if very core application things were broken (unable to access internet, svelte stores are broken)